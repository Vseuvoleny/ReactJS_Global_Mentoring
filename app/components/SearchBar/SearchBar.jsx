import React, { forwardRef, useState, useCallback } from "react";
import Button from "../../elements/Button/Button";
import Input from "../../elements/Input/Input";
import "./SearchBar.module.scss";

const SearchBar = forwardRef((props, ref) => {
  const [param, setSaram] = useState();

  const showFilms = async (e) => {
    e.preventDefault();
    navigate(param);
  };

  const filmRequest = useCallback(showFilms, [param]);

  return (
    <div className="search-bar">
      <h2 className="search-bar__title">find your movie</h2>
      <form className="search-bar__search-container" onSubmit={filmRequest}>
        <Input
          ref={ref}
          classes="input-field"
          placeholder="What do you want to watch?"
          withLabel={false}
          value={param}
          onChange={(e) => setSaram(e.target.value)}
        />
        <Button className="search-bar" type="submit" text="Search" />
      </form>
    </div>
  );
});

export default SearchBar;
