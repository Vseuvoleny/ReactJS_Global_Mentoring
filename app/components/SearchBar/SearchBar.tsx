import React, { forwardRef, useState, useCallback } from "react";
import Button from "../../elements/Button/Button";
import Input from "../../elements/Input/Input";
import classes from "./SearchBar.module.scss";

const SearchBar = forwardRef((_, ref) => {
  const [param, setSaram] = useState();

  const showFilms = async (e) => {
    e.preventDefault();
  };

  const filmRequest = useCallback(showFilms, [param]);

  return (
    <div className={classes.search_bar}>
      <h2 className={classes.search_bar__title}>find your movie</h2>
      <form
        className={classes.search_bar__search_container}
        onSubmit={filmRequest}
      >
        <Input
          ref={ref}
          classes={classes.input_field}
          placeholder="What do you want to watch?"
          withLabel={false}
          value={param}
          onChange={(e) => setSaram(e.target.value)}
        />
        <Button type="submit" text="Search" />
      </form>
    </div>
  );
});

export default SearchBar;
