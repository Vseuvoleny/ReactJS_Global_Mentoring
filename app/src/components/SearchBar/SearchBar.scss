.search-bar {
  &__title {
    font-size: 40px;
    font-weight: 300;
    letter-spacing: 1px;
    line-height: 49px;
    text-transform: uppercase;
  }

  &__search-container {
    display: flex;
    margin-top: 40px;
  }

  .input-container.input-field {
    width: 80%;
    margin-right: 14px;
  }
  .input-field {
    background: rgba(50, 50, 50, 0.8);
    border: 0;
    border-radius: 4px;
    color: #fff;
    font-size: 20px;
    line-height: 24px;
    outline: none;

    &::placeholder {
      color: rgba($color: #fff, $alpha: 0.3);
      font-size: 20px;
      line-height: 24px;
    }
  }

  &__button {
    background: #f65261;
    border: 0;
    border-radius: 4px;
    color: #fff;
    cursor: pointer;
    font-size: 20px;
    line-height: 24px;
    outline: none;
    padding: 18px 74px;
    text-transform: uppercase;
  }
}
