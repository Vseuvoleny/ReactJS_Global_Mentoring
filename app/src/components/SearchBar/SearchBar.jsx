import React, { forwardRef, useState, useCallback } from "react";
import { useDispatch } from "react-redux";
import { getFilms } from "../../Store/ActionCreator";
import { useParams, useNavigate } from "react-router-dom";
import Button from "../../Elements/Button/Button";
import Input from "../../Elements/Input/Input";
import "./SearchBar.scss";

const SearchBar = forwardRef((props, ref) => {
  const [param, setSaram] = useState(useParams().queryParams);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const showFilms = async (e) => {
    e.preventDefault();
    const URL = `http://localhost:4000/movies?sortOrder=desc&search=${param}&searchBy=title&limit=6`;
    const res = await fetch(URL);
    const { data } = await res.json();
    navigate(param);
    dispatch(getFilms(data));
  };

  const filmRequest = useCallback(showFilms, [param]);

  return (
    <div className="search-bar">
      <h2 className="search-bar__title">find your movie</h2>
      <form className="search-bar__search-container" onSubmit={filmRequest}>
        <Input
          ref={ref}
          classes="input-field"
          placeholder="What do you want to watch?"
          withLabel={false}
          value={param}
          onChange={(e) => setSaram(e.target.value)}
        />
        <Button className="search-bar" type="submit" text="Search" />
      </form>
    </div>
  );
});

SearchBar.displayName = "SearchBar";
export default SearchBar;
