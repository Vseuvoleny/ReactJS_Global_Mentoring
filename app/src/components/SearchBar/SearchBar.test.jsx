import React from "react";
import { render } from "@testing-library/react";
import SearchBar from "./SearchBar";
import { BrowserRouter } from "react-router-dom";

describe("Searchbar component", () => {
  test("should render SearchBar component", () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SearchBar />
      </BrowserRouter>
    );
    const button = getByRole("button", { name: /search/i });
    expect(button).toBeInTheDocument();
    expect(getByText(/find your movie/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
