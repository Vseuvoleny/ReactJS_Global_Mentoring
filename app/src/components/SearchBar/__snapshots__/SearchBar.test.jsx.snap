// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar component should render SearchBar component 1`] = `
<div>
  <div
    class="search-bar"
  >
    <h2
      class="search-bar__title"
    >
      find your movie
    </h2>
    <form
      class="search-bar__search-container"
    >
      <div
        class="input-container input-field"
      >
        <input
          autocomplete="off"
          class="input input-field"
          data-testid="input"
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
      </div>
      <div
        class="search-bar"
      >
        <button
          class="search-bar__button filled"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</div>
`;
